types/types.go

```diff
@@ -381,9 +424,18 @@ type Muter interface {
 	Mutes(model.LabelSet) bool
 }
 
-// TimeMuter determines if alerts should be muted based on the specified current time and active time interval on the route.
+// TimeMuter determines if alerts should be muted or not for active and mute
+// time intervals based on the specified current time.
 type TimeMuter interface {
```

Comment:
This is where I made changes to the interface based on the discussion in #3562. I'm not super happy with how this interface has turned out though as there are now two methods `Active` and `Mutes` which look like inverses of each other when they are not.

For example, it looks like if I pass a series of mute time intervals to `Active` and it returns false, then  passing the same mute time intervals to `Mute` should return true. That is, `Active` and `Mute` are negations of each other. However, in this case both will instead return true, as `Active` is only supposed to be used with active time intervals and `Mute` is only supposed to be used with mute time intervals. I have attempted to document their behavior as such, but I'm still not happy with it.